.github/workflows/build.yml

```
@@ -25,48 +25,54 @@ on:
     types: [published]
 
 jobs:
-  build:
-    runs-on: ubuntu-latest
+  build_wheels:
+    name: Build Wheels
+    runs-on: ${{ matrix.os }}
     strategy:
+      fail-fast: false
       matrix:
-        python-version: ["3.9", "3.10", "3.11"]
+        build: ["cp39", "cp310", "cp311"]
+        os: [ubuntu-latest, macos-latest]
 
     steps:
     - name: Checkout
       uses: actions/checkout@v4
 
-    - name: Build tfx-bsl
-      id: build-tfx-bsl
-      uses: ./.github/reusable-build
-      with:
-        python-version: ${{ matrix.python-version }}
+    - name: Build wheels
+      uses: pypa/cibuildwheel@v2.23.3
+      env:
+        CIBW_BUILD: ${{ matrix.build }}-*
 
+    - uses: actions/upload-artifact@v4
+      with:
+        name: cibw-wheels-${{ matrix.os }}-${{ matrix.build }}-${{ strategy.job-index }}
+        path: ./wheelhouse/*.whl
 
   upload_to_pypi:
     name: Upload to PyPI
     runs-on: ubuntu-latest
-    if: (github.event_name == 'release' && startsWith(github.ref, 'refs/tags'))
-    needs: [build]
+    #if: (github.event_name == 'release' && startsWith(github.ref, 'refs/tags')) || (github.event_name == 'workflow_dispatch')
```

Sorry, I should have called this out specifically before. I think we need to remove the `#` here, we don't want to be uploading to PyPI on every workflow run:

```suggestion
    if: (github.event_name == 'release' && startsWith(github.ref, 'refs/tags')) || (github.event_name == 'workflow_dispatch')
```